### PR TITLE
One Script To Rule Them All (or at least the really simple ones)

### PR DESCRIPTION
**Pull request by maschall**
I wanted to test my coffeescript skills, and thought this might be better way of doing all the simple scripts.

I'm going with the simple approach, if we want to remove a key, than we set the percent to 0%.

What ya think?

If this is ok, I will expand this to be able to take multiple responses with a random.

Also this will accept regex for the key so "(damn it|damnit|dammit|damnit) jeff (kelley|kelly)" can map to I don't get it

### CODE REVIEW

File: scripts/ostrta.coffee [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  
+loadAllCommands = ( robot, command ) ->
+  keys = fetch( robot, brainKey(command, key, "Keys") )
+  if keys
+    for key in keys
+      tellRobotWhatToDo( robot, command, key )
+  
+module.exports = (robot) ->
+  commands = _.keys(commandMap)
+  for command in commands
```
Comment (nwest) [inline scripts/ostrta.coffee]: Why not chain a _.map here instead of a for?

Comment (maschall) [inline scripts/ostrta.coffee]: i feel like map is good for getting an object back out after iterating over the list vs. just iterating over a list and calling a method.

**Comment by maschall** [comment]
Oh and I do have the list of commands ready to load up hipchat with the old phrases

File: scripts/ostrta.coffee (hunk 1 above)
Comment (nwest) [inline scripts/ostrta.coffee]: Ah, good point. Didn't realize that was a void return.

File: scripts/ostrta.coffee [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  robot.brain.set(key, value)
+  
+fetch = (robot, key) ->
+  robot.brain.get(key)
+  
+percentFromString = (percentString) ->
+  parseInt(percentString, 10)
+  
+tellRobotWhatToDo = (robot, command, key) ->
+  robot[commandMap[command]](new RegExp("(#{key}(\\s|$))", 'i'), (msg) ->
```
Comment (nwest) [inline scripts/ostrta.coffee]: Is there a possibility of someone doing something nasty here? (think injection)

File: scripts/ostrta.coffee [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    for key in keys
+      tellRobotWhatToDo( robot, command, key )
+  
+module.exports = (robot) ->
+  commands = _.keys(commandMap)
+  for command in commands
+    loadAllCommands( robot, command )
+
+  commandsRegex = commands.join('|')
+  robot.respond new RegExp("(#{commandsRegex}) (\"[^\"]+\"|[^\\s]+) ([\\d]+%\\s)?(.+)$"), (msg) ->
```
Comment (nwest) [inline scripts/ostrta.coffee]: Should we break this regex down into named parts? Might be easier to read for future devs.

**Comment by nwest** [comment]
The only thing that worries me is that if redis dies somehow, we'll lose all these. It might be useful to look into redis backup.

File: scripts/ostrta.coffee (hunk 3 above)
Comment (maschall) [inline scripts/ostrta.coffee]: named parts makes sense, how do you get the named parts out of the match object in coffeescript?

File: scripts/ostrta.coffee (hunk 2 above)
Comment (maschall) [inline scripts/ostrta.coffee]: thats an interesting thought, I guess I could try something fun out.

File: scripts/ostrta.coffee (hunk 3 above)
Comment (nwest) [inline scripts/ostrta.coffee]: something like http://stackoverflow.com/a/12317411/518071?

Comment (maschall) [inline scripts/ostrta.coffee]: oh i was thinking something like:
(?<command>#{commandsRegex}) (?<key>\"[^\"]+\"|[^\\s]+) (?<percent>[\\d]+%\\s)?(?<response>.+)$

but it appears that has not made it into javascript or coffeescript: https://github.com/jashkenas/coffeescript/issues/1378

File: scripts/ostrta.coffee (hunk 2 above)
Comment (maschall) [inline scripts/ostrta.coffee]: so I was thinking if you could close of the " in the regex early than you probably could inject something, however the input regex would capture the ".  So I tried sending in the ascii char and a unicode char, but hipchat sends those as just plain characters.  I don't think you can close off the " in the regexp constructor, but I could also be dead wrong.

**Comment by maschall** [comment]
alternatively we could also use a hipchat bot called Hearsay to do all these things too:
image

**Comment by macklinu** [comment]
Part of me will miss all of these small scripts, even though they all technically repeat code. I think it's nice to see a code history of scripts that are added and remember what was happening at Labs at the time to inspire those scripts (i.e. qfax). Merging this script may have a negative effect on that history as well as general contribution to Hubot. Maybe I'm overthinking it; what do you think?

**Comment by maschall** [comment]
I had the same thoughts too. (I know Jenai said she wanted to learn to write a script to do one of the small things, and I don't want to take away that either.) 

I was only thinking there are common complaints like the frequency of responses and also trying to make sure we live clean code in everything we do.

**Comment by macklinu** [comment]
@maschall I definitely see the merits in both things you said. I think we just need to integrate this script into hubot, see how we all like it, and assess from there!

**Comment by maschall** [comment]
ok, maybe I should un remove the old scripts?

**Comment by macklinu** [comment]
Perhaps we use the new system for a bit and then revert 59ca09c9ab81c3d1e18d44227e7846c4ec0c6442 if people want individual scripts back?

**Pull request merged**